homeassistant/config_entries.py

```diff
@@ -1706,6 +1725,10 @@ async def async_finish_flow(
             self.config_entries._async_clean_up(existing_entry)  # noqa: SLF001
 
         result["result"] = entry
+        if not existing_entry:
```

Comment:
This could go before or after adding the config entry and cleanups.
**Before**: Stop the flow if `next_flow` is not correct and doesn't add the main config entry
**After**: The main config entry gets created and then the user gets an error that the next flow could not be loaded.

Reply:
Realized it needs to go after, since it needs to add the main config entry before continuing.
Possibly we could remove it, if validation fails, but that seems a bit wasteful as process.